src/main/common/utils/numericUtils.ts

```diff
@@ -10,7 +10,19 @@ export function toNumberOrUndefined (value: any): number {
   }
 
   const strVal: string = value && value.toString().trim()
-  const numberVal: number = strVal.length > 0 ? _.toNumber(strVal) : undefined
 
-  return isNaN(numberVal) ? undefined : numberVal
+  // checks length, followed by a filter by not containing ',' or valid thousand separators
+  const numberVal: number = strVal.length > 0
+    ? (!strVal.includes(',') || containsAThousandSeparator(strVal))
+      ? numeral(strVal).value()
+      : strVal
+    : undefined
+
+  // returns undefined for
+  return !numberVal ? undefined : numberVal
+}
+
+function containsAThousandSeparator (input: string): boolean {
+  const matchedArray = input.match(/^[1-9]\d{0,2}(\,\d{3})+(\.\d*)?$/)
+  return matchedArray ? matchedArray.length > 0 : false
```

Comment:
I didn't check the return value of `match` method, apologies. In this case I think we could do this to make the code simpler:
```typescript
const THOUSANDS_SEPARATORS_REGEXP = /^[1-9]\d{0,2}(\,\d{3})+(\.\d*)?$/
return THOUSANDS_SEPARATORS_REGEXP.test(input.trim())
```